src/tasks/harbor/harbor.ts

```diff
@@ -98,29 +105,83 @@ function setAuth(secret): void {
   robotApi.setDefaultAuthentication(bearerAuth)
 }
 
+async function getProjectId(team: string): Promise<string> {
+  const project = (await doApiCall(errors, `Get project for team ${team}`, () =>
+    projectsApi.getProject(team),
+  )) as Project
+  if (!project) return ''
+  const projectId = `${ensure(project.projectId)}`
+  return projectId
+}
+
 // NOTE: assumes OIDC is not yet configured, otherwise this operation is NOT possible
-async function createRobotSecret(): Promise<RobotSecret> {
+async function createSystemRobotSecret(): Promise<RobotSecret> {
   const { body: robotList } = await robotApi.listRobot()
-  const existing = robotList.find((i) => i.name === `robot$${robot.name}`)
+  const existing = robotList.find((i) => i.name === `robot$${systemRobot.name}`)
   if (existing?.id) {
     const existingId = existing.id
-    await doApiCall(errors, `Deleting previous robot account ${robot.name}`, () => robotApi.deleteRobot(existingId))
+    await doApiCall(errors, `Deleting previous robot account ${systemRobot.name}`, () =>
+      robotApi.deleteRobot(existingId),
+    )
   }
   const { id, name, secret } = await doApiCall(
     errors,
-    `Create robot account ${robot.name} with system level perms`,
-    () => robotApi.createRobot(robot),
+    `Create robot account ${systemRobot.name} with system level perms`,
+    () => robotApi.createRobot(systemRobot),
   )
   const robotSecret: RobotSecret = { id, name, secret }
-  await createSecret(secretName, namespace, robotSecret)
+  await createSecret(systemSecretName, systemNamespace, robotSecret)
   return robotSecret
 }
 
-async function ensureSecret(): Promise<RobotSecret> {
-  let robotSecret = (await getSecret(secretName, namespace)) as RobotSecret
+async function createProjectRobotSecret(team: string): Promise<RobotSecret> {
+  const projectId = await getProjectId(team)
+
+  const projectRobot: any = {
+    name: projectRobotName,
+    duration: -1,
+    description: 'Used by kubernetes to pull images from harbor in each team',
+    disable: false,
+    level: 'project',
+    permissions: [
+      {
+        kind: 'project',
+        namespace: team,
+        access: [
+          {
+            resource: 'repository',
+            action: 'pull',
+          },
+        ],
+      },
+    ],
+  }
+
+  const { body: robotList } = await robotv1Api.listRobotV1(projectId)
+  const existing = robotList.find((i) => i.name === `robot$${team}+${projectRobot.name}`)
+
+  if (existing?.id) {
+    const existingId = existing.id
+    await doApiCall(errors, `Deleting previous robot account ${existing.name}`, () =>
+      robotv1Api.deleteRobotV1(projectId, existingId),
```

Comment:
Can we not just use the robotApi? Instead of the V1? Because we then can omit all robotV1Api calls right?

Reply:
Yes, but I couldn't get the list of project robots with robotApi and it worked with robotApiV1.